src/web_sys.rs

```diff
@@ -2570,18 +2582,30 @@ impl HasContext for Context {
                             ty,
                             offset as i32,
                         ),
-                    PixelUnpackData::Slice(slice) => gl
-                        .tex_sub_image_2d_with_i32_and_i32_and_u32_and_type_and_opt_u8_array(
-                            target,
-                            level,
-                            x_offset,
-                            y_offset,
-                            width,
-                            height,
-                            format,
-                            ty,
-                            Some(slice),
-                        ),
+                    PixelUnpackData::Slice(data) => {
```

Comment:
Thank you! There is [some similar logic in `tex_image_2d` already](https://github.com/grovesNL/glow/pull/134/files#diff-a624050e2e701b14057c0b81e01edfb3eb98d6dec91cc7770fcb26f0820703f0R1803) that includes some additional types that can be used here (the complete table is in https://www.khronos.org/registry/webgl/specs/latest/2.0/#TEXTURE_PIXELS_TYPE_TABLE), so I wonder if we could combine these somehow?

I noticed we seem to be missing this for `tex_image_3d` and `tex_sub_image_3d` too. We could land this PR first if you'd like though

Reply:
Good call. I've moved the logic from `tex_image_2d` into it's own function and am using that function from `tex_image_2d`, `tex_sub_image_2d`, `tex_image_3d`, and `tex_sub_image_3d`.